orchard_key_components.py

```diff
@@ -37,6 +38,14 @@ def diversify_hash(d):
         P = group_hash(b'z.cash:Orchard-gd', b'')
     return P
 
+def prf_nf_orchard(nk, rho):
+    return poseidon_hash(nk, rho)
+
+def derive_nullifier(nk, rho: Fp, psi: Fp, cm):
+    scalar = to_base(prf_nf_orchard(nk, rho)) + psi  # addition mod p
+    point = NULLIFIER_K_BASE * to_scalar(scalar) + cm
```

Comment:
Similarly, `scalar` is not a byte array so we shouldn't call `leos2ip` on it, as `to_scalar` does. It should be something like `Scalar(scalar.s)` I think.